src/engine/providers/utils/open-ai-api.ts

```diff
@@ -0,0 +1,153 @@
+import type OpenAI from 'openai';
+import type { ChatCompletionChunk } from 'openai/resources/index.mjs';
+import type { Message, ModelResponse, ModelResponseStream } from '../../inference.js';
+import { estimateInputTokens, estimateOutputTokens } from '../../tokenizer.js';
+import { responseStyles, type ProviderConfig } from '../config.js';
+
+export async function getChatCompletion(
+  api: OpenAI,
+  config: ProviderConfig,
+  messages: Message[],
+): Promise<ModelResponse> {
+  const systemMessage: Message = {
+    role: 'system',
+    content: config.systemPrompt,
+  };
+
+  const startTime = performance.now();
+  const response = await api.chat.completions.create({
+    messages: [systemMessage, ...messages],
+    model: config.model,
+    ...responseStyles[config.responseStyle],
+  });
+  const responseTime = performance.now() - startTime;
```

Comment:
some kind of failure handling would be nice

Reply:
There should be a top level error handling. Basically without inference provider response we cannot do much more than quit